sockeye/decoder.py

```diff
@@ -81,20 +79,17 @@ def decode_sequence(self,
 
     @abstractmethod
     def decode_step(self,
-                    target_embed: mx.sym.Symbol,
-                    target_embed_lengths: mx.sym.Symbol,
-                    target_embed_max_length: int,
+                    step: int,
                     target_embed_prev: mx.sym.Symbol,
                     source_encoded_max_length: int,
                     *states: mx.sym.Symbol) -> Tuple[mx.sym.Symbol, mx.sym.Symbol, List[mx.sym.Symbol]]:
         """
-        Decodes a single time step given the embedded target sequence and previous decoder states.
+        Decodes a single time step given the current step, the previous embedded target word,
+        and previous decoder states.
         Returns decoder representation for the next prediction, attention probabilities, and next decoder states.
         Implementations can maintain an arbitrary number of states.
 
-        :param target_embed: Embedded target sequence. Shape: (batch_size, target_embed_max_length, target_num_embed).
-        :param target_embed_lengths: Lengths of embedded target sequences. Shape: (batch_size,).
-        :param target_embed_max_length: Size of embedded target sequence dimension.
+        :param step: Global step of inference procedure.
```

Comment:
could you add a note whether it's 0 or 1 based?

Reply:
I can :)